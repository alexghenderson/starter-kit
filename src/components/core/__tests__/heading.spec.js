import React from 'react';
import enzyme from 'enzyme';
import Heading from '../heading';

describe('Heading core component', () => {
  it('should render with default props', () => {
    const wrapper = enzyme.mount(<Heading>test-content</Heading>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render alternate tags', () => {
    const wrapper = enzyme.mount(<Heading tag="p">test-p</Heading>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should adapt tag to heading level', () => {
    const wrapper = enzyme.mount(<Heading level={2}>test-level-2</Heading>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render different levels', () => {
    const wrapper = enzyme.mount(
      <React.Fragment>
        {[1, 2, 3, 4].map(level => (
          <Heading level={level} key={level}>
            text-{level}
          </Heading>
        ))}
      </React.Fragment>
    );
    expect(wrapper).toMatchSnapshot();
  });
});
