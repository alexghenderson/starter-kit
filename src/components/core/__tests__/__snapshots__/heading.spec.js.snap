// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading core component should adapt tag to heading level 1`] = `
.emotion-0 {
  font-size: 1.5rem;
  line-height: 1.33;
  -webkit-letter-spacing: -0.7px;
  -moz-letter-spacing: -0.7px;
  -ms-letter-spacing: -0.7px;
  letter-spacing: -0.7px;
  font-family: "Lato",sans-serif;
}

<h2
  className="emotion-0"
>
  test-level-2
</h2>
`;

exports[`Heading core component should render alternate tags 1`] = `
.emotion-0 {
  font-size: 1.75rem;
  line-height: 1.29;
  -webkit-letter-spacing: -1.6px;
  -moz-letter-spacing: -1.6px;
  -ms-letter-spacing: -1.6px;
  letter-spacing: -1.6px;
  font-family: "Lato",sans-serif;
}

<p
  className="emotion-0"
>
  test-p
</p>
`;

exports[`Heading core component should render different levels 1`] = `
Array [
  .emotion-0 {
  font-size: 1.75rem;
  line-height: 1.29;
  -webkit-letter-spacing: -1.6px;
  -moz-letter-spacing: -1.6px;
  -ms-letter-spacing: -1.6px;
  letter-spacing: -1.6px;
  font-family: "Lato",sans-serif;
}

<h1
    className="emotion-0"
  >
    text-
    1
  </h1>,
  .emotion-0 {
  font-size: 1.5rem;
  line-height: 1.33;
  -webkit-letter-spacing: -0.7px;
  -moz-letter-spacing: -0.7px;
  -ms-letter-spacing: -0.7px;
  letter-spacing: -0.7px;
  font-family: "Lato",sans-serif;
}

<h2
    className="emotion-0"
  >
    text-
    2
  </h2>,
  .emotion-0 {
  font-size: 1.25rem;
  line-height: 1.4;
  -webkit-letter-spacing: -0.6px;
  -moz-letter-spacing: -0.6px;
  -ms-letter-spacing: -0.6px;
  letter-spacing: -0.6px;
  font-family: "Lato",sans-serif;
}

<h3
    className="emotion-0"
  >
    text-
    3
  </h3>,
  .emotion-0 {
  font-size: 1rem;
  line-height: 1.25;
  -webkit-letter-spacing: -0.6px;
  -moz-letter-spacing: -0.6px;
  -ms-letter-spacing: -0.6px;
  letter-spacing: -0.6px;
  font-family: "Lato",sans-serif;
}

<h4
    className="emotion-0"
  >
    text-
    4
  </h4>,
]
`;

exports[`Heading core component should render with default props 1`] = `
.emotion-0 {
  font-size: 1.75rem;
  line-height: 1.29;
  -webkit-letter-spacing: -1.6px;
  -moz-letter-spacing: -1.6px;
  -ms-letter-spacing: -1.6px;
  letter-spacing: -1.6px;
  font-family: "Lato",sans-serif;
}

<h1
  className="emotion-0"
>
  test-content
</h1>
`;
